tests/unit/test_markdown.py

```diff
@@ -225,6 +227,22 @@ def should_generate_env_prefix_and_nested_delimiter(runner: CliRunner, mocker: M
         assert "`prefix_sub_model__nested`" in actual_string
         assert "`prefix_sub_model__deep__leaf`" in actual_string
 
+    @staticmethod
+    def should_generate_settings_with_settings_sub_model(runner: CliRunner, mocker: MockerFixture):
+        actual_string = run_app_with_settings(mocker, runner, SettingsWithSettingsSubModel)
+
+        assert "`prefix_direct`" in actual_string
+        assert "`prefix_sub_model_nested`" in actual_string
+        assert "`prefix_sub_model_sub_model_2__leaf`" in actual_string
+
+    @staticmethod
+    def should_generate_settings_with_settings_sub_model_no_prefix_or_delimiter(
+        runner: CliRunner, mocker: MockerFixture
+    ):
+        actual_string = run_app_with_settings(mocker, runner, SettingsWithSettingsSubModelNoPrefixOrDelimiter)
+
+        assert "`my_leaf_leaf`" in actual_string
```

Comment:
This is actually incorrect. The top level is ignored.

```suggestion
        assert "`leaf`" in actual_string
```

Reply:
I'm not sure I understand. This is the output that gets produced:

```
# `my_leaf_leaf`

**required**
```

Reply:
Yes, but that is not what `pydantic` accepts in the environment variables. It loads the value from `LEAF`, not `MY_LEAF_LEAF`.

Reply:
Huh. I wonder why I thought that would work?